cwms-data-api/src/main/java/cwms/cda/data/dto/location/kind/ProjectStructure.java

```diff
@@ -20,34 +20,34 @@
 
 package cwms.cda.data.dto.location.kind;
 
+import com.fasterxml.jackson.annotation.JsonIgnore;
 import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
 import com.fasterxml.jackson.databind.PropertyNamingStrategies;
 import com.fasterxml.jackson.databind.annotation.JsonNaming;
-import cwms.cda.api.errors.FieldException;
-import cwms.cda.api.errors.RequiredFieldException;
-import cwms.cda.data.dto.CwmsDTOBase;
+import cwms.cda.data.dto.CwmsDTO;
+import cwms.cda.data.dto.CwmsDTOValidator;
 import cwms.cda.data.dto.CwmsId;
 import cwms.cda.data.dto.Location;
-import java.util.ArrayList;
-import java.util.List;
 
 @JsonInclude(JsonInclude.Include.NON_NULL)
 @JsonNaming(PropertyNamingStrategies.KebabCaseStrategy.class)
-public abstract class ProjectStructure implements CwmsDTOBase {
+public abstract class ProjectStructure extends CwmsDTO {
+    @JsonProperty(required = true)
```

Comment:
Can the validator find the fields annotated as "required" without them also having to be specified in validateInternal?

Reply:
Maybe a default validate function on CwmsDTOBase that scans for the annotations?